CMakeLists.txt

```diff
@@ -13,6 +13,8 @@ TRIBITS_PACKAGE_DECL(DataTransferKit)
 TRIBITS_ADD_DEBUG_OPTION()
 TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()
 
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+
```

Comment:
No, don't get me wrong, I agree with you in that we should not hard-code this. I thought you had a question of how it works. I'm pretty sure it won't work on every compiler, as some of them have pretty exotic flags.